panda/src/audio/audioSound.h

```diff
@@ -89,13 +90,15 @@ class EXPCL_PANDA_AUDIO AudioSound : public TypedReferenceCount {
   // return: playing time in seconds.
   virtual PN_stdfloat length() const = 0;
 
-  // Controls the position of this sound's emitter.  px, py and pz are the
+  // Controls the position, velocity and direction of this sound's emitter.  px, py and pz are the
   // emitter's position.  vx, vy and vz are the emitter's velocity in UNITS
-  // PER SECOND (default: meters).
+  // PER SECOND (default: meters). dx, dy and dz are the emitter's direction
   virtual void set_3d_attributes(PN_stdfloat px, PN_stdfloat py, PN_stdfloat pz,
-                                 PN_stdfloat vx, PN_stdfloat vy, PN_stdfloat vz);
-  virtual void get_3d_attributes(PN_stdfloat *px, PN_stdfloat *py, PN_stdfloat *pz,
-                                 PN_stdfloat *vx, PN_stdfloat *vy, PN_stdfloat *vz);
+      PN_stdfloat vx, PN_stdfloat vy, PN_stdfloat vz,
+      PN_stdfloat dx = LVector3::forward().get_x(), PN_stdfloat dy = LVector3::forward().get_y(), PN_stdfloat dz = LVector3::forward().get_z());
+  virtual void get_3d_attributes(PN_stdfloat* px, PN_stdfloat* py, PN_stdfloat* pz,
+      PN_stdfloat* vx, PN_stdfloat* vy, PN_stdfloat* vz,
+      PN_stdfloat* dx, PN_stdfloat* dy, PN_stdfloat* dz);
```

Comment:
You changed the signature of a virtual method, but that requires changing the signature of all implementations of this method in derived classes.  As-is this breaks FMOD etc.

get_3d_attributes either needs an overload with fewer args for compatibility or you need to default these to nullptr with an extra check in the method to not write them if they're nullptr.

It may be less work to just make a separate set/get_3d_direction if you want.  You can also make it accept a direction vector if you do that rather than 3 separate args so you can handle the LVector3::forward default more cleanly.